base/loading.jl

```diff
@@ -490,155 +435,128 @@ end
-        for line in eachline(io)
-            if occursin(re_array_of_tables, line)
-                uuid == where && break
-                uuid = deps = nothing
-                state = :stanza
-            elseif state === :stanza
-                if (m = match(re_uuid_to_string, line)) !== nothing
-                    uuid = UUID(m.captures[1])
-                elseif (m = match(re_deps_to_any, line)) !== nothing
-                    deps = String(m.captures[1])
-                elseif occursin(re_subsection_deps, line)
-                    state = :deps
-                elseif occursin(re_section, line)
-                    state = :other
-                end
-            elseif state === :deps && uuid == where
-                # [deps] section format gives both name and uuid
-                if (m = match(re_key_to_string, line)) !== nothing
-                    m.captures[1] == name && return UUID(m.captures[2])
+    d = parsed_toml(cache, manifest_file)
+    found_where = false
+    found_name = false
+    for (dep_name, entries) in d
+        for entry in entries
+            haskey(entry, "uuid") || continue
+            if UUID(entry["uuid"]) == where
+                found_where = true
+                # deps is either a list of names (deps = ["DepA", "DepB"]) or
+                # a table of entries (deps = {"DepA" = "6ea...", "DepB" = "55d..."}
+                deps = get(entry, "deps", nothing)
+                if deps isa Vector
+                    found_name = name in deps
+                    break
+                elseif deps isa Dict
+                    for (dep, uuid) in deps
+                        if dep === name
+                            return PkgId(UUID(uuid), name)
+                        end
+                    end
                 end
             end
         end
-        # now search through `deps = []` string to see if we have an entry for `name`
-        uuid == where || return false
-        deps === nothing && return true
-        # TODO: handle inline table syntax
-        if deps[1] != '[' || deps[end] != ']'
-            @warn "Unexpected TOML deps format:\n$deps"
-            return false
-        end
-        occursin(repr(name), deps) || return true
-        seekstart(io) # rewind IO handle
-        # finally, find out the `uuid` associated with `name`
-        return something(manifest_file_name_uuid(io, name), false)
     end
-    found_or_uuid isa UUID && return PkgId(found_or_uuid, name)
-    found_or_uuid && return PkgId(name)
-    return nothing
+    found_where || return nothing
+    found_name || return PkgId(name)
+    # Only here is deps was not a dict which mean we have a unique name for the dep
+    if !haskey(d, name) || length(d[name]) != 1
+        error("expected a single entry for $(repr(name)) in $(repr(project_file))")
+    end
+    entry = first(d[name])
+    if found_name
+        uuid = get(entry, "uuid", nothing)
+        return PkgId(UUID(uuid), name)
```

Comment:
`ERROR: MethodError: Cannot `convert` an object of type Nothing to an object of type UInt128`